lib/src/main/java/com/diffplug/spotless/JarState.java

```diff
@@ -37,6 +37,13 @@
  * catch changes in a SNAPSHOT version.
  */
 public final class JarState implements Serializable {
+
+	private static ClassLoader OVERRIDE_CLASS_LOADER = null;
```

Comment:
@nedtwigg / @jbduncan This here is one of the spots, where the `cli` needs to change things in order to be able to compile to a nativeImage -> loading stuff dynamically via JarState (e.g. choosable google-java-format) is not possible. Things need to be pinned down at build time. So here I took a nasty shortcut to just override the class loader in JarState - that is definitively not the way to go.

To actually allow the `cli` to change the strategy for retrieving a class loader (which contains classes that have been burned into the nativeImage) here, maybe we could use Java's `ServiceLoader` approach? That way maven/gradle plugin can keep using the flexible and dynamic approach, while `cli` could provide an implementation, that returns a fixed ClassLoader.

Reply:
to clarify: something along the lines of the current solution is fine?

Reply:
Exactly, what you have is great!